lightningd/gossip_control.c

```diff
@@ -59,6 +59,34 @@ static void got_txout(struct bitcoind *bitcoind,
 	tal_free(scid);
 }
 
+static void got_filteredblock(struct bitcoind *bitcoind,
+		      struct filteredblock *fb,
+		      void *arg)
+{
+	struct short_channel_id *scid = (struct short_channel_id *)arg;
```

Comment:
Seems like it's actually a typesafe cb, so you can make scid an arg?

Reply:
That's what I was going for as well, but if I have the correct type in there I get this:

```
lightningd/gossip_control.c: In function ‘get_txout’:
ccan/ccan/typesafe_cb/typesafe_cb.h:33:2: error: passing argument 3 of ‘bitcoind_getfilteredblock_’ from incompatible pointer type [-Werror=incompatible-pointer-types]
  __builtin_choose_expr(      \
  ^
ccan/ccan/typesafe_cb/typesafe_cb.h:109:2: note: in expansion of macro ‘typesafe_cb_cast’
  typesafe_cb_cast(rtype (*)(__VA_ARGS__, atype),   \
  ^~~~~~~~~~~~~~~~
lightningd/bitcoind.h:162:8: note: in expansion of macro ‘typesafe_cb_preargs’
        typesafe_cb_preargs(void, void *, \
        ^~~~~~~~~~~~~~~~~~~
lightningd/gossip_control.c:122:3: note: in expansion of macro ‘bitcoind_getfilteredblock’
   bitcoind_getfilteredblock(topo->bitcoind, short_channel_id_blocknum(scid), got_filteredblock, scid);
   ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from lightningd/gossip_control.c:1:0:
lightningd/bitcoind.h:154:6: note: expected ‘void (*)(struct bitcoind *, struct filteredblock *, void *)’ but argument is of type ‘void (*)(struct bitcoind *, struct filteredblock *, struct short_channel_id *)’
 void bitcoind_getfilteredblock_(struct bitcoind *bitcoind, u32 height,
      ^~~~~~~~~~~~~~~~~~~~~~~~~~
```

and I got stuck debugging macro expansions. Maybe you have a good idea what is happening?

Reply:
```
void bitcoind_getfilteredblock_(struct bitcoind *bitcoind, u32 height,
				void (*cb)(struct bitcoind *bitcoind,
					   struct filteredblock *fb,
					   void *arg),
				void *arg);
#define bitcoind_getfilteredblock(bitcoind_, height, cb, arg)		\
	bitcoind_getfilteredblock_((bitcoind_),				\
				   (height),				\
				   typesafe_cb_preargs(void, void *,	\
						       (cb), (arg),	\
						       struct bitcoind *, \
						       const struct filteredblock *), \
				   (arg))
```

You convert the fn if it takes a `const struct filteredblock *`, but it doesn't.  Fixed it to take a const everywhere, because we can.